ynput/OpenPype
# Global: persistent staging directory for renders

## Brief description
Allows configure if staging directory (`stagingDir`) should be persistent with use of profiles.

## Description
With this feature, users can specify a transient data folder path based on presets, which can be used during the creation and publishing stages. In some cases, these DCCs automatically add a rendering path during the creation stage, which is then used in publishing.

One of the key advantages of this feature is that it allows users to take advantage of faster storages for rendering, which can help improve workflow efficiency. Additionally, this feature allows users to keep their rendered data persistent, and use their own infrastructure for regular cleaning.

However, it should be noted that some productions may want to use this feature without persistency. Furthermore, there may be a need for retargeting the rendering folder to faster storages, which is also not supported at the moment.

It is studio responsibility to clean up obsolete folders with data.

Location of the folder is configured in `project_anatomy/templates/others`. ('transient' key is expected, with 'folder' key, could be more templates)

Which family/task type/subset is applicable is configured in:
`project_settings/global/tools/publish/transient_dir_profiles`

## Additional info
During publishing of renders, they are first rendered to work dir then publishing process will create reviews and thumbnails and only after that all of it is copied/moved to publish folder.

With renders, it will be much safer to render them to some common staging/transient folder and do the publishing from there. This way artists/other processes won't corrupt yet unpublished data so easily.

As example in Maya:
Renders in Maya are now done from published scene. We could change image prefix only in published scene to point to that staging directory, so farm will render there but scene that artist has open will point still to work dir.

## Testing notes:
1. Check configuration and modify folder path for transient stagingDir in `project_anatomy/templates/others`
![2023-03-16 17_24_43-Window](https://user-images.githubusercontent.com/4457962/225686773-bdae0ba9-0f28-476c-9c7f-b0fb3b1c6bc7.png)

3. Configure profiles in `project_settings/global/tools/publish/custom_staging_dir_profiles`
![2023-03-21 18_23_54-Window](https://user-images.githubusercontent.com/4457962/226691684-aee5dd8e-abfc-410c-84fd-b2e044ff6be1.png)

5. Publish and check that temporary files are in specific location configured in step 1
